lib/src/ast.dart

```diff
@@ -27,47 +45,69 @@ class TextQuery implements Query {
 /// Phrase query to match "[text]" for a list of words inside quotes.
 class PhraseQuery extends TextQuery {
   final List<TextQuery> children;
-  PhraseQuery(String phrase, this.children) : super(phrase, isExactMatch: true);
+  const PhraseQuery({
+    required super.text,
+    required this.children,
+    required super.startIndex,
+    required super.endIndex,
+  }) : super(isExactMatch: true);
 
   @override
   String toString({bool debug = false}) =>
       '"' + children.map((n) => n.toString(debug: debug)).join(' ') + '"';
 }
 
 /// Scopes [child] [Query] to be applied only on the [field].
-class FieldScope implements Query {
-  final String field;
+class FieldScope extends Query {
+  final TextQuery field;
   final Query child;
 
-  FieldScope(this.field, this.child);
+  const FieldScope({
+    required this.field,
+    required this.child,
+    required super.startIndex,
+    required super.endIndex,
+  });
 
   @override
   String toString({bool debug = false}) =>
       '$field:${child.toString(debug: debug)}';
 }
 
 /// Describes a [field] [operator] [text] tripled (e.g. year < 2000).
-class FieldCompareQuery implements Query {
-  final String field;
-  final String operator;
+class FieldCompareQuery extends Query {
+  final TextQuery field;
+  final TextQuery operator;
```

Comment:
Is the change from `String` to `TextQuery` for the `operator` field correct?

Reply:
It could help figure out where the operator is.

e.g. all of these are valid:
```
field=value
field = value
field    =    value
```

Reply:
I think this should be a separate `FieldOperator extends Query` type, with `text` or `value` + `position`. The `TextQuery` has `isExactMatch` which doesn't make sense for this field, and it doesn't feel right. wdyt?

Reply:
Does it make sense to have `FieldCompareQuery.field`, `FieldCompareQuery .operator` and `FieldScope.operator` all share the same class. e.g. something like

```dart
class FieldElementQuery extends Query {
  final String text;
  const FieldElementQuery({
    required this.text,
    required super.position,
  });

  @override
  String toString({bool debug = false}) =>
      _debug(debug, text);
}
```

Reply:
Or Alternatively

```dart
class SimpleTextQuery extends Query {
  final String text;
  const SimpleTextQuery({
    required this.text,
    required super.position,
  });

  @override
  String toString({bool debug = false}) => _debug(debug, text);
}

/// Text query to match [text].
///
/// [isExactMatch] is set when the [text] was inside quotes.
class TextQuery extends SimpleTextQuery {
```

Reply:
Or... Remove `isExactMatch` from TextQuery and have it exclusively for `PhraseQuery`?